projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -1,15 +1,22 @@
 import { ChangeDetectionStrategy, Component } from '@angular/core';
 import { Product } from '@spartacus/core';
 import { CmsComponentData } from '@spartacus/storefront';
-import { Observable, of } from 'rxjs';
+import { EMPTY, Observable, of } from 'rxjs';
 import {
   distinctUntilKeyChanged,
   filter,
   map,
   switchMap,
+  withLatestFrom,
 } from 'rxjs/operators';
 import { CmsMerchandisingCarouselComponent } from '../../../cds-models/cms.model';
 import { CdsMerchandisingProductService } from '../../facade/cds-merchandising-product.service';
+import { MerchandisingProducts } from '../../model/merchandising-products.model';
+
+export interface MerchandisingCarouselModel {
```

Comment:
Since this interface is not going to be re-used elsewhere, maybe we could avoid creating it and just define the type inline:
```ts
merchandisingCarouselModel$: Observable<
    {metadata: Map<string, string>;
  items$: Observable<Product>[];}
  > = this.componentData$.pipe(
...
```

Reply:
The same could be done for the mapping method.

Reply:
Sure, works for me